# Source Code, Cara Lain: Untuk Menyelesaikan Soal Iris dengan Metode Algorima Naive Bayes Classifier

In [2]:
from sklearn import datasets
from sklearn.naive_bayes import GaussianNB
from sklearn.model_selection import train_test_split  # Menggunakan modul yang diperbarui
from sklearn.metrics import accuracy_score, classification_report
import pandas as pd

# Memuat dataset
data = datasets.load_iris()

# Mengonversi ke DataFrame
df = pd.DataFrame(data.data, columns=data.feature_names)
df['target'] = data.target

# Membagi dataset menjadi set pelatihan dan pengujian
X_train, X_test, y_train, y_test = train_test_split(df.drop('target', axis=1), df['target'], test_size=0.3, random_state=42)

# Inisialisasi model
model = GaussianNB()

# Melatih model
model.fit(X_train, y_train)

# Memprediksi pada set pengujian
y_pred = model.predict(X_test)

# Mengevaluasi model
print("Akurasi:", accuracy_score(y_test, y_pred))
print("Laporan Klasifikasi:")
print(classification_report(y_test, y_pred))


Akurasi: 0.9777777777777777
Laporan Klasifikasi:
              precision    recall  f1-score   support

           0       1.00      1.00      1.00        19
           1       1.00      0.92      0.96        13
           2       0.93      1.00      0.96        13

    accuracy                           0.98        45
   macro avg       0.98      0.97      0.97        45
weighted avg       0.98      0.98      0.98        45



## Kesimpulan :
## Dari Perhitungan akurasi diatas dapat dilihat bahwa skor akurasi model denga Naive Bayes ini adalah 0.97 atau 97%
